warden/x/act/types/v1beta1/action.go

```diff
@@ -63,3 +71,28 @@ func (a *Action) AddApprover(ctx sdk.Context, address string) error {
 
 	return nil
 }
+
+func (a *Action) AddVote(ctx sdk.Context, participant string, voteType ActionVoteType) error {
+	if a.Status != ActionStatus_ACTION_STATUS_PENDING {
+		return errors.Wrapf(ErrInvalidActionStatus, "can't add a vote to an action that's not pending")
+	}
+
+	for _, v := range a.Votes {
+		if v.Participant == participant {
+			return ErrAlreadyParticipatedInVoting
```

Comment:
i think we should also allow an user to change their mind, i don't see any particular reason for making the vote immutable